src/repr/src/antichain.rs

```diff
@@ -0,0 +1,27 @@
+// Copyright Materialize, Inc. and contributors. All rights reserved.
+//
+// Use of this software is governed by the Business Source License
+// included in the LICENSE file.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0.
+
+use mz_proto::{RustType, TryFromProtoError};
+use timely::progress::Antichain;
+
+use crate::Timestamp;
+
+include!(concat!(env!("OUT_DIR"), "/mz_repr.antichain.rs"));
+
+impl RustType<ProtoU64Antichain> for Antichain<Timestamp> {
+    fn into_proto(&self) -> ProtoU64Antichain {
+        ProtoU64Antichain {
+            elements: self.elements().to_vec(),
```

Comment:
this is where you'll have to flatten out the `&[Timestamp]` into `Vec<u8>` in your PR (and unflatten a few lines down)